vsintegration/src/FSharp.Editor/CodeFix/FixIndexerAccess.fs

```diff
@@ -70,9 +70,16 @@ type internal FsharpFixRemoveDotFromIndexerAccessOptIn() as this =
     override _.FixableDiagnosticIds = Seq.toImmutableArray fixableDiagnosticIds
 
     override _.RegisterCodeFixesAsync context : Task =
-        backgroundTask {
-            let relevantDiagnostics = this.GetPrunedDiagnostics(context)
+        backgroundTask { this.RegisterFix(CodeFix.RemoveIndexerDotBeforeBracket, title, context, TextChange(context.Span, "")) }
 
-            if not relevantDiagnostics.IsEmpty then
-                this.RegisterFix(CodeFix.RemoveIndexerDotBeforeBracket, title, context, TextChange(context.Span, ""))
-        }
+    override this.GetFixAllProvider() =
+        FixAllProvider.Create(fun fixAllCtx doc allDiagnostics ->
+            backgroundTask {
+                let changes =
+                    allDiagnostics |> Seq.map (fun x -> TextChange(x.Location.SourceSpan, ""))
+
+                let! text = doc.GetTextAsync(fixAllCtx.CancellationToken)
+
+                CodeFixHelpers.reportCodeFixRecommendation allDiagnostics doc CodeFix.RemoveIndexerDotBeforeBracket
```

Comment:
We should have this consolidated instead of copypasting to 8 places. 

CodeFixHelpers already work as a single point of the code fix flow for most of the fixes so it would be awesome to have bulk fixes flowing through one place as well.

Reply:
It is already consolidated by calling a shared function - this is how composition works.
Over-DRY for the sake of avoiding 1 functional call would be harmful here.

Reply:
The same way as they would know to make 1 special call into a helper module => that would not change a bit.
Just here the composition uses a function and is very explicit about what it does, instead of hiding behaviours.

Also, having the call explicit makes sure it is called at the right time.

(e.g. with respect to telemetry, it is only when a document is really requested for preview, not when codefix is just calculated but not actioned on)

Reply:
Because the bulk codefix API does not have any boilerplate ceremony to encapsulate. There is neither a CodeAction, nor a codefix registration for a context happening.
It just returns a Task<Document>, that is it.

If someone wants to a attach a new cross-cutting concern, probably the cleanest place would be decorating FixAllProvider.Create from Roslyn.
I can show that as an example in this PR, so that new features can built upon it.

Reply:
The current version is pretty much what I meant, this looks fine to me.